src/txn/txn.c

```diff
@@ -140,42 +140,43 @@ __wt_txn_release_snapshot(WT_SESSION_IMPL *session)
 
 /*
  * __wt_txn_user_active --
- *     Check whether there are any running user transactions. Note that a new transactions may start
- *     on a session we have already examined and the caller needs to be aware of this limitation.
- *     Exclude prepared user transactions from this check.
+ *     Check whether there are any running user transactions. Note that a new transaction may start
+ *     after we return from this call and therefore caller should be aware of this limitation.
  */
 bool
 __wt_txn_user_active(WT_SESSION_IMPL *session)
 {
     WT_CONNECTION_IMPL *conn;
     WT_SESSION_IMPL *session_in_list;
+    WT_TXN_GLOBAL *txn_global;
+    WT_TXN_SHARED *txn_shared;
     uint32_t i, session_cnt;
     bool txn_active;
 
     conn = S2C(session);
     txn_active = false;
+    txn_global = &conn->txn_global;
+
+    /* We're going to scan the table: wait for the lock. */
+    __wt_readlock(session, &txn_global->rwlock);
```

Comment:
The read lock doesn't prevent the transactions that are starting in parallel to the check that is performed here. We need to take the write lock.

Reply:
Fixed in 8c3a76b